### PR TITLE
refactor: improve parameter render component

### PR DESCRIPTION
**Pull request by anovazzi1**
This pull request includes several refactorings and additions to the parameter render component.

The refactorings include:

- Updating the dynamic parameter in the ParameterRenderComponent to false

- Updating the CodeAreaComponent to use CodeAreaModal for the node toolbar

- Updating the RefreshParameterComponent to use InputProps

- Updating the InputProps in parameterRenderComponent/types.ts

- Updating the parameterRenderComponent/types.ts to use handleOnNewValue instead of onChange

- Updating the TableModal to support disabled state

- Updating the parameterRenderComponent/types.ts to include TableComponentType

- Updating the parameterRenderComponent to use TableNodeComponent from the correct folder

- Updating the codeAreaComponent imports and props in parameterRenderComponent

- Updating the parameterRenderComponent/types.ts to include CodeAreaComponentType

- Removing unused imports of CodeAreaComponent in nodeToolbarComponent

- Updating the ToggleComponentType in components/index.ts

- Updating the ToggleShadComponent to handle new value changes

- Moving the InputFileComponent to parameterRenderComponent/components folder

- Adding the FileComponentType interface to parameterRenderComponent/types.ts

- Updating the InputFileComponent props in parameterRenderComponent

- Moving the PromptAreaComponent to parameterRenderComponent/components folder

- Updating the PromptAreaComponent to handle new value changes

- Moving the LinkComponent to parameterRenderComponent/components folder

- Adding the LinkComponentType interface to parameterRenderComponent/types.ts

- Updating the LinkComponent to use InputProps in parameterRenderComponent

- Moving the KeypairListComponent to parameterRenderComponent/components folder

- Adding the KeyPairListComponentType interface to parameterRenderComponent/types.ts

- Updating the KeypairListComponent to use InputProps in parameterRenderComponent

- Moving the DictComponent to parameterRenderComponent/components folder

- Updating the KeyPairListComponentType interface in parameterRenderComponent/types.ts

- Updating the KeypairListComponent to use InputProps in parameterRenderComponent

- Moving the InputListComponent to parameterRenderComponent/components folder

- Adding the StrRenderComponentType interface to parameterRenderComponent/types.ts

- Updating the InputListComponent to use InputProps in parameterRenderComponent

- Moving the IOFieldView to IOModal/components folder

- Moving the DropdownComponent to parameterRenderComponent/components folder

- Adding the DropDownComponentType interface to parameterRenderComponent/types.ts

- Updating the DropdownComponent import in StrRenderComponent

- Updating the StrRenderComponent to use baseInputProps

- Updating the StrRenderComponent to use TextAreaComponent from correct location

- Removing unused imports

- Adding the TextAreaComponentType interface to parameterRenderComponent/types.ts

- Updating the TextAreaComponent to use handleOnNewValue instead of onChange

- Updating the StrRenderComponent to use baseInputProps

- Moving the inputGlobalComponent to parameterRenderComponent/components folder

- Updating the InputGlobalComponent to use handleOnNewValue instead of onChange

- Updating the StrRenderComponent to use baseInputProps

- Moving the multiselectComponent to parameterRenderComponent/components folder

- Moving the MultiselectComponent to parameterRenderComponent/components folder

- Removing unused code in StrRenderComponent

- Updating the FloatComponent to use InputProps

The additions include:

- Adding the EmptyParameterComponent to render empty parameters

These changes improve code organization, maintainability, and functionality of the parameter render component.

### CODE REVIEW

**Review by Cristhianzl — APPROVED** [review]
lgtm

File: src/frontend/src/components/parameterRenderComponent/components/dictComponent/index.tsx [hunk 1]
```diff
@@ -1,40 +1,37 @@
...
   editNode = false,
   id = "",
-}: DictComponentType): JSX.Element {
-  // Create a reference to the value
-
+}: InputProps<object | object[] | string>): JSX.Element {
   useEffect(() => {
     if (disabled) {
-      onChange({});
+      handleOnNewValue({ value: {} });
```
Comment (lucaseduoli) [inline src/frontend/src/components/parameterRenderComponent/components/dictComponent/index.tsx:18]: I think skipSnapshot is necessary here as well.

File: src/frontend/src/components/parameterRenderComponent/components/keypairListComponent/index.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
   id,
-}: KeyPairListComponentType): JSX.Element {
+}: InputProps<
+  object[] | object | string,
+  KeyPairListComponentType
+>): JSX.Element {
   useEffect(() => {
     if (disabled && value.length > 0 && value[0] !== "") {
-      onChange([{ "": "" }]);
+      handleOnNewValue({ value: [{ "": "" }] });
```
Comment (lucaseduoli) [inline src/frontend/src/components/parameterRenderComponent/components/keypairListComponent/index.tsx:26]: I think skipSnapshot is necessary here as well.

File: src/frontend/src/components/parameterRenderComponent/components/linkComponent/index.tsx
Comment (lucaseduoli) [inline src/frontend/src/components/parameterRenderComponent/components/linkComponent/index.tsx:1]: I think there is a missing cleaning of the component value, like the other ones

File: src/frontend/tests/core/unit/tableInputComponent.spec.ts [hunk 3]
```diff
@@ -26,6 +27,14 @@ test("user must be able to interact with table input component", async ({
   const secondRandomText = Math.random().toString(36).substring(7);
   const thirdRandomText = Math.random().toString(36).substring(7);
 
+  const getUA = await page.evaluate(() => navigator.userAgent);
+  const userAgentInfo = uaParser(getUA);
+  let control = "Meta";
```
Comment (lucaseduoli) [inline src/frontend/tests/core/unit/tableInputComponent.spec.ts:32]: shouldn't it be Ctrl?

**Review by lucaseduoli — CHANGES_REQUESTED** [review]
LGTM, only these minor changes

File: src/frontend/tests/core/unit/tableInputComponent.spec.ts (hunk 3 above)
Comment (Cristhianzl) [reply src/frontend/tests/core/unit/tableInputComponent.spec.ts]: yes

File: src/frontend/src/components/parameterRenderComponent/components/linkComponent/index.tsx
Comment (anovazzi1) [reply src/frontend/src/components/parameterRenderComponent/components/linkComponent/index.tsx:1]: this one the user does not change the link value directly, only using the component code, so the disabled clean up does not make sense in this case

**Review by lucaseduoli — APPROVED** [review]
LGTM!

**Pull request merged**